jerry-core/parser/js/js-lexer.c

```diff
@@ -2626,8 +2626,14 @@ lexer_expect_object_literal_id (parser_context_t *context_p, /**< context */
   if (create_literal_object)
   {
 #if ENABLED (JERRY_ES2015)
-    if (is_class_method && lexer_compare_literal_to_string (context_p, "constructor", 11))
+    if ((is_class_method || (ident_opts & LEXER_OBJ_IDENT_CLASS_GENERATOR))
+        && lexer_compare_literal_to_string (context_p, "constructor", 11))
```

Comment:
It seems to me this prevents non-class methods with a "constructor" name to be created.

Reply:
This new flag is only used while parsing class literals so it should not prevent something like that. Added a new test case too.

Reply:
But the `||` says that it is is_class_method OR generator.
What happens if you run this code:

`function *constructor() {}`

Reply:
Or if this code is only used when you parse a class, then `(ident_opts & LEXER_OBJ_IDENT_CLASS_GENERATOR)` cannot be true unless is_class_method is true, so the whole check is unnecessary.